### PR TITLE
ngoc: update ibc v6

### PR DESCRIPTION
**Pull request by minhngoc274**
## 1. Summary
Fixes # (issue)

## 2.Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## 5. Checklist

- [ ] Does the Readme need to be updated?

### CODE REVIEW

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>bf16724</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1422.app-dev.quicksilver.zone'>https://pr-1422.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/FWo9ebDv6YX8GThTYrzbi9c3Lnpv'>https://vercel.com/quicksilver/quicksilver/FWo9ebDv6YX8GThTYrzbi9c3Lnpv</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8652621229)

File: app/app.go [hunk 1]
```diff
@@ -3,20 +3,15 @@ package app
...
 	"encoding/json"
-	"io"
-	"net/http"
-	"os"
-	"path/filepath"
-	"strings"
-
-	"github.com/CosmWasm/wasmd/x/wasm"
-	wasmkeeper "github.com/CosmWasm/wasmd/x/wasm/keeper"
-	"github.com/prometheus/client_golang/prometheus"
```
Comment (joe-bowman) [inline app/app.go:14]: I don’t think this ought to be removed?

Comment (joe-bowman) [reply app/app.go:14]: (Specifically the Prometheus import)

**Review by joe-bowman — CHANGES_REQUESTED** [review]
Fundamentally looks good.

is the change from sdk.Msg to protoMessage required by the IBC v6 upgrade?

is removing the Prometheus import going to change any behavior wrt to metrics? Presumably it was there for a reason?

File: app/app.go (hunk 1 above)
Comment (faddat) [reply app/app.go:14]: Yes we should keep Prometheus

**Review by faddat — APPROVED** [review]
This is good to go for me, as soon as we figure out what to do with Prometheus.

We should also make sure to get the store key deletion into the upgrade handler and follow IBC v5 to v6 migration docs.

**Comment by minhngoc274** [comment]
> Fundamentally looks good.
> 
> is the change from sdk.Msg to protoMessage required by the IBC v6 upgrade?
> 
> is removing the Prometheus import going to change any behavior wrt to metrics? Presumably it was there for a reason?

Yes, it is required to change from sdk.Msg to proto.Message

**Comment by minhngoc274** [comment]
I will looking for the Prometheus remove

File: app/app.go (hunk 1 above)
Comment (hoank101) [reply app/app.go:14]: in this case Prometheus only used in wasm

**Review by joe-bowman — APPROVED** [review]
lgtm

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>caecf66</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1422.app-dev.quicksilver.zone'>https://pr-1422.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/Ao4aj4WCD3s8Cjo4vwNfiQ12v6xV'>https://vercel.com/quicksilver/quicksilver/Ao4aj4WCD3s8Cjo4vwNfiQ12v6xV</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8680451708)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>aebf21a</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1422.app-dev.quicksilver.zone'>https://pr-1422.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/Eak5TtnPDgzzo4oJVS7Ph8LA1wgH'>https://vercel.com/quicksilver/quicksilver/Eak5TtnPDgzzo4oJVS7Ph8LA1wgH</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8680471305)

**Comment by tropicaldog** [comment]
@joe-bowman after applying the refactor I already done from the sdk47 branch (https://github.com/quicksilver-zone/quicksilver/pull/1422/commits/1a97cd578daefd2784d2f5511066f78b5171ae03), the `make test` command is now passing.

Could you please take a look and see if it is good? I'm not sure if the code actually need refactor, or we should refactor [test setup ](https://github.com/quicksilver-zone/quicksilver/blob/main/x/interchainstaking/keeper/keeper_test.go#L122) following v5->v6 integration instead?

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>d4b05bc</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1422.app-dev.quicksilver.zone'>https://pr-1422.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/3S7V8ZocnoQwEnUqZ7ZjsSVGn9yb'>https://vercel.com/quicksilver/quicksilver/3S7V8ZocnoQwEnUqZ7ZjsSVGn9yb</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8683158593)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>593d06a</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1422.app-dev.quicksilver.zone'>https://pr-1422.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/A2tL3wR4dynvS8uDU7rFe7Fdw2Hd'>https://vercel.com/quicksilver/quicksilver/A2tL3wR4dynvS8uDU7rFe7Fdw2Hd</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8683179485)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>49612dc</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1422.app-dev.quicksilver.zone'>https://pr-1422.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/9eYxH9nRTszXTdvHR8eQDBpVz8R9'>https://vercel.com/quicksilver/quicksilver/9eYxH9nRTszXTdvHR8eQDBpVz8R9</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8683941875)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>dc333aa</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1422.app-dev.quicksilver.zone'>https://pr-1422.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/HqqMp4C3MoXG2J5JRck26ARd4BQb'>https://vercel.com/quicksilver/quicksilver/HqqMp4C3MoXG2J5JRck26ARd4BQb</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8684820340)

**Pull request merged**